src/Microsoft.TestPlatform.ObjectModel/Host/ITestRunTimeProvider.cs

```diff
@@ -65,8 +65,9 @@ public interface ITestRuntimeProvider
         /// Launches the test host for discovery/execution.
         /// </summary>
         /// <param name="testHostStartInfo">Start parameters for the test host.</param>
+        /// <param name="cancellationToken"></param>
         /// <returns>ProcessId of launched Process. 0 means not launched.</returns>
-        Task<int> LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo);
+        Task<int> LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, CancellationToken cancellationToken);
```

Comment:
If Runtime provider need to keep additional information, then it can store that info locally. As for lauchasync it's job it to return whether host was successfully launched or not.